docs/dev/azure/azure_client_certs_auth.md

```diff
@@ -0,0 +1,37 @@
+# Azure Authentication using Client certificates
+
+Azure accepts client certificates as a means of authentication in the service principal and terraform accepts it too. 
+The installer now supports authentication via the certs after [1] merged. 
+
+### Pitfalls
+Although the installer can now use the certs to authenticate, CCO does not support this and hence the installer
+should create the cluster in manual credentials mode only.
+
+### Prerequisites
+- A certificate that is suitable for Azure is created.
```

Comment:
I think this would be the helpful part to explain. Is there an easy way to do this?

For example, can we use `az ad sp create-for-rbac --create-cert` to achieve this? If so it would be helpful to include what is the easy way to get started doing this.